### ISSUE TITLE
main/main.c has logical mistakes

### ISSUE THREAD

**Issue by leleliu008** [universal-ctags/ctags#2996]
1. [main/main.c#L125-L164](https://github.com/universal-ctags/ctags/blob/master/main/main.c#L125-L164)
```
#elif defined (HAVE__FINDFIRST)

#if defined (HAVE__FINDFIRST)
#endif

#endif
```

2. [main/main.c#L91-L164](https://github.com/universal-ctags/ctags/blob/master/main/main.c#L91-L164)
```
#if defined (HAVE_OPENDIR) && (defined (HAVE_DIRENT_H) || defined (_MSC_VER))
        static bool recurseUsingOpendir (const char *const dirName){}
#elif defined (HAVE__FINDFIRST)
        static bool createTagsForWildcardEntry (){}
        static bool createTagsForWildcardUsingFindfirst (const char *const pattern){}
#endif
```
here should change to following:
```
#if defined (HAVE_OPENDIR) && (defined (HAVE_DIRENT_H) || defined (_MSC_VER))
        static bool recurseUsingOpendir (const char *const dirName){}
#endif

#if defined (HAVE__FINDFIRST)
        static bool createTagsForWildcardEntry (){}
        static bool createTagsForWildcardUsingFindfirst (const char *const pattern){}
#endif
```
I find on `MSYS2/MINGW32` and `MSYS2/MINGW64` have `opendir()` function, but don't define `HAVE_OPENDIR` in `./main/e_msoft.h`, so `#if defined (HAVE_OPENDIR) && (defined (HAVE_DIRENT_H) || defined (_MSC_VER))` always `false`, 

When I use CMake build system, I checked `opendir()` function in build script, and defined `HAVE_OPENDIR` macro, so the logic error.

**Comment by masatake** [comment]
Please, make a pull request for the solution that you found:-)

**Comment by masatake** [comment]
Run 'git blame' if you find a code questionable?
I have changed many areas in main/. However, some areas have not been changed from Exuberant Ctags.
If a questionable area has not been changed since the Exuberant Ctags, and if you know the way to improve,
in many cases, opening a pull request is better.

If a questionable area has been changed by @masatake, opening an issue makes sense.

**Comment by leleliu008** [comment]
@masatake OK.

**Comment by k-takata** [comment]
This will be fixed in #3072.
